[develop] Fix python path for slurm config unit test in cookbook

NSsirena opened the pull request:
### Description of changes
* Align tox python path for slurm unit tests to the one used by the recipes
* Extract common functions in `config_utils.py`
* Extract Queue and ComputResource Renderers in `config_renderer.py`
* Review logic to identify jinja template folder

### Tests
* All unit tests launched from the IDE and with tox are passing

### References
* [Code refactoring](https://github.com/aws/aws-parallelcluster-cookbook/pull/1955)
* [Custom Slurm Settings Renderer](https://github.com/aws/aws-parallelcluster-cookbook/pull/1962)

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

NSsirena committed:
Fix python path for slurm unit tests

NSsirena committed:
Extracts config_utils file

NSsirena committed:
Extracts config_renderer file

NSsirena committed:
Moves _get_template_folder into config_utils

NSsirena committed:
Fix code-linters

jdeamicis commented on test/unit/slurm/test_slurm_config_generator.py:
```diff
@@ -9,13 +9,19 @@
+def _get_template_folder() -> str:
+    module_file = inspect.getsourcefile(generate_slurm_config_files)
+    template_path = Path(module_file).parent / "templates"
+    return str(template_path)
```
Is this a solid implementation to retrieve the templates folder? I'm asking because I honestly don't know.

Still, I guess it should be equivalent to the old `os.path.dirname(slurm.__file__) + "/templates"`

NSsirena replied:
It is equivalent "by design" :)

It is also a solid implementation and it is more general since starts from the path of a module containing a given function. As of now since we only have one use I embedded the function name, and so the module.
However if other cookbooks need a similar function it could be moved in common and passing the function as parameter it would work also for other paths/modules/cookbooks if they are in the python_path of the code that is invoking the function.

jdeamicis commented on cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/config_utils.py:
```diff
@@ -0,0 +1,98 @@
+import math
+from typing import List, Tuple
+
+def _get_instance_types(compute_resource_config) -> List[str]:
```
Nitpick (maybe addressed later?): these are now functions used by external scripts/classes, so let's drop the leading underscore.

NSsirena replied:
Good catch.
I fixed ` get_template_folder()` but I forgot the other ones since I moved them in bulk.

Fixing.

jdeamicis commented on cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/config_utils.py:
```diff
@@ -1,6 +1,15 @@
 from typing import List, Tuple
 
+
+def get_template_folder() -> str:
```
Maybe let's call this function something like `get_slurm_config_template_folder()`?

NSsirena replied:
(See comment above) I kept it general since the function can easily be generalized (by simply making the function name a parameter) and used in a more general context.
I didn't do this now, since we do not yet need it.

jdeamicis reviewed (commented):
Good job, I'm just leaving some comments to improve the readability of the code.

NSsirena committed:
Removes underscore from function names

Utility functions moved in config_utils are now used
externally so the underscore in the function names is
misleading.
